htpy/__init__.py

```diff
@@ -1,456 +1,36 @@
-        context_value = (context or {}).get(self.context, self.context.default)
-
-        if context_value is _NO_DEFAULT:
-            raise LookupError(
-                f'Context value for "{self.context.name}" does not exist, '  # pyright: ignore
-                f"requested by {self.debug_name}()."
-            )
-        return _iter_chunks_node(self.func(context_value), context)  # pyright: ignore
-
-    def encode(self, encoding: str = "utf-8", errors: str = "strict") -> bytes:
-        return str(self).encode(encoding, errors)
-
-
-class _NO_DEFAULT:
-    pass
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class Context(t.Generic[T]):
-    name: str
-    _: dataclasses.KW_ONLY
-    default: T | type[_NO_DEFAULT] = _NO_DEFAULT
-
-    def provider(self, value: T, node: Node) -> ContextProvider[T]:
-        return ContextProvider(self, value, node)
-
-    def consumer(
-        self,
-        func: Callable[t.Concatenate[T, P], Node],
-    ) -> Callable[P, ContextConsumer[T]]:
-        @functools.wraps(func)
-        def wrapper(*args: P.args, **kwargs: P.kwargs) -> ContextConsumer[T]:
-            return ContextConsumer(self, func.__name__, lambda value: func(value, *args, **kwargs))
-
-        return wrapper
-
-
-@deprecated(  # type: ignore[misc,unused-ignore]
-    "iter_node is deprecated and will be removed in a future release. "
-    "Please use the .iter_chunks() method on elements/fragments instead."
-)  # pyright: ignore [reportUntypedFunctionDecorator]
-def iter_node(x: Node) -> Iterator[str]:
-    return fragment[x].iter_chunks()
-
-
-def _iter_chunks_node(x: Node, context: Mapping[Context[t.Any], t.Any] | None) -> Iterator[str]:
-    while not isinstance(x, BaseElement) and callable(x):
-        x = x()
-
-    if x is None:
-        return
-
-    if x is True:
-        return
-
-    if x is False:
-        return
-
-    if hasattr(x, "iter_chunks"):
-        yield from x.iter_chunks(context)  # pyright: ignore
-    elif isinstance(x, str | _HasHtml):
-        yield str(_escape(x))
-    elif isinstance(x, int):
-        yield str(x)
-    elif isinstance(x, Iterable) and not isinstance(x, _KnownInvalidChildren):  # pyright: ignore [reportUnnecessaryIsInstance]
-        for child in x:
-            yield from _iter_chunks_node(child, context)
-    else:
-        raise TypeError(f"{x!r} is not a valid child element")
-
-
-@functools.lru_cache(maxsize=300)
-def _get_element(name: str) -> Element:
-    if not name.islower():
-        raise AttributeError(
-            f"{name} is not a valid element name. html elements must have all lowercase names"
-        )
-    return Element(_python_to_html_name(name))
+__all__: list[str] = [
+    "Attribute",
+    "BaseElement",
+    "Context",
+    "ContextConsumer",
+    "ContextProvider",
+    "Element",
+    "Fragment",
+    "HTMLElement",
+    "Node",
+    "Renderable",
+    "VoidElement",
+    "comment",
+    "fragment",
+    "iter_node",
+    "render_node",
```

Comment:
Should we keep `__all__` empty and instead use the `from foo import bar as bar` trick to import and re-export these?

Reply:
I did not really think much about setting __all__ to [] but I agree it make more sense to keep it like that.

`from htpy import *` feels like it can be very confusing since we use the module level `__getattr__` to create elements dynamically.

I pushed a commit to clear __all__.